geoips/xarray_utils/data.py

```diff
@@ -416,6 +407,98 @@ def sector_xarray_spatial(
     return sector_xarray
 
 
+def get_minmax_latlon_from_area_def(area_def, lon_pad, lat_pad):
+    """Retrieve the Min/Max Lat/Lon from the provided area_def.
+
+    Given a PyResample AreaDefinition Object, retrieve the minimum and maximum
+    latitude and longitude values that should be used to sector the xarray efficiently.
+
+    If the provided area def encapsulates either of the poles, set the min/max latitude
+    value to -90/90 degrees accordingly, and match the other latitude value to what was
+    provided in the area_def bounds.
+
+    For longitudes, if the sector doesn't contain a pole, wrap them to ensure they exist
+    in the 0-360 range, otherwise set to -180, 180.
+
+    Parameters
+    ----------
+    area_def: PyResample AreaDefinition()
+        - The area_definition that defines the bounds of the data we will be sectoring.
+    lon_pad: float
+        - The amount (in degrees) of padding to be added to the longitude bounds.
+    lat_pad: float
+        - The amount (in degrees) of padding to be added to the latitude bounds.
+
+    Returns
+    -------
+    min_lon, min_lat, max_lon, max_lat: float
+        - These degree (float) values will be returned in the order listed above.
+    """
+    extent_lonlat = list(
+        area_def.area_extent_ll
+    )  # [min_lon, min_lat, max_lon, max_lat]
+    min_lon, max_lon = -180, 180
+
+    # Check if the area_def includes either North or South Pole. If it does, we have to
+    # manually adjust the lat_lon extent to reflect the correct extent for both lat/lon.
+    # Otherwise go about business as usual.
```

Comment:
You're right. All sectors that were handled correctly before will be handled the same as before. Only sectors that intersect the poles will be handled differently.

I'm 99% sure this will be fine for all sectors, even one that kind of weirdly wraps around the pole.